BUILD.zstd

```diff
@@ -0,0 +1,223 @@
+cc_library(
```

Comment:
A straight conversion of https://github.com/facebook/zstd/blob/dev/lib/BUCK

Seems to work on Linux and macOS per iostreams_test below, at least for basic cases.